README.md

```diff
@@ -22,24 +22,103 @@ We recognise the following major repositories and organisations may create `<rep
 ### Package list schema
 
 Package lists are formatted as JSON.
-Each file is a list of package entries with the following fields.
+The schema for major repositories is formally described in [`schema.json`](./schema.json).
+This is a [JSON Schema](https://json-schema.org/overview/what-is-jsonschema) document using the 2020-12 draft.
+
+In brief, each file is a list of package entries with the following fields.
 
 - `package_name` (text, case sensitive, uniquely defines a package for a repository)
-- `version` (text, [PEP440 style](https://peps.python.org/pep-0440/#version-specifiers))
+- `version` (text, comparison operator plus [PEP440 style](https://peps.python.org/pep-0440/#version-specifiers) version number)
```

Comment:
Is it necessary to include the comparison operator when the approval applies to a specific version? Might be normal to do that for Python, but not for R.

Reply:
I think so, at least we need some unambiguous way to define versions so I picked PEP440 as one such method.

That feels easier than making our own, or also adding "no comparison means `==`".

Reply:
I'd have assumed that if a version is stated as approved, it is that version that has been approved. 

If the comparison operator is always `==`, is it adding anything? Or do we want to allow people to use other comparison operators?

Part of the reason I ask is that if I wanted to pull the version numbers as well as the package names, so as to only install the specifically approved versions, I'd then have to strip the comparison operator out when installing a specific version for R. It's not exactly a big problem - I'd also probably have to do some text manipulation to construct the right URL anyway.

I'm not strongly against it.

Reply:
That would be one way of doing it. However, I can see the problem there being that maintaining a list will require adding (and revoking) all minor releases of a package. I don't think anyone would be able to commit to that.

There will also be organisations (including our own) where we don't approve specific releases. The example lists already use other comparison operators.

Reply:
@craddm Do you think this is OK _for now_?

I think long term we are quite likely to find the PEP440 doesn't match very well with non-Python repositories. We may want schema for each major repository type. However, I think we can fix that when it arises and what we have here is enough for a proof of concept.